proposals/NNNN-add-customdebugdescription-conformance-to-anykeypath.md

```diff
@@ -0,0 +1,137 @@
+2. the linker has stripped the symbol names we're trying to look up
+
+In these cases, we would print the following: 
+
+#### Offset case
+`<offset [x] ([typename])>` where `x` is the memory offset we read from the reflection metadata, and `typename` is the type that will be returned. 
+So 
+```
+print(\Theme.backgroundColor) // outputs "\Theme.<offset 0 (Color)>"
+```
+
+#### `lookupSymbol` failure case
+
+In this case we'll print the address in memory as hex, plus the type name: 
+```
+print(\Theme.overlay) // outputs \Theme.<computed 0xABCDEFG (Color)>
+```
+
+As it might be difficult to correlate a memory address with the name of the function, the type name may be useful here to provide extra context. 
+
+## Source compatibility
+
+Programs that extend `AnyKeyPath` to implmenent `CustomDebugStringConvertible` themselves will no longer compile. 
+
+Calling `print` on a KeyPath will of course produce different results. 
+
+It is unlikely that any existing Swift program is depending on the existing behavior in a production context. While it is likely that someone, somewhere has written code in unit tests that depends on the output of this function, any issues that result will be easy for the authors of such code to identify and fix, and will likely represent an improvement in the readability of those tests. 
+
+## Effect on ABI stability
+
+This proposal will add a new conformance to an existing protocol to an existing type in the stdlib.  
+
+## Effect on API resilience
+
+This change could break existing Swift programs that add their own conformance to `AnyKeyPath`. 
+
+However, it seems unlikely that any Swift program would rely on `debugDescription` in production, when the `description` property is far more appropriate for such a purpose. 
+
+
+## Alternatives considered
+
+### Print fully qualified names or otherwise add more information to the output
+
+ex. `\ModuleName.MyType.myField`, `<KeyPath<MyType, MyFieldType>> \ModuleName.MyType.myField` `(writable) \Theme.backgroundColor`
+
+As this is just for debugging, it seems likely that the information currently being provided would be enough to resolve any ambiguities. If ambiguities arose during a debugging session, in most cases the user could figure out exactly which keypath they were dealing with simply by running `po myKeyPath == \MyType.myField` till they found the right one. 
+
+### Modify KeyPaths to include a string description
+
+This is an obvious solution to this problem, and would likely be very easy to implement, as the compiler already produces `_kvcString`. 
+
+It has the additional advantage of being 100% reliable, to the point where it arguably could be the basis for implementing `description` rather than `debugDescription`. 
+
+However, it would add to the code size of the compiled code, perhaps unacceptably so. Furthermore, it precludes the possiblity of someday printing out the arguments of subscript based keypaths, as
```

Comment:
```suggestion
However, it would add to the code size of the compiled code, perhaps unacceptably so. Furthermore, it precludes the possibility of someday printing out the arguments of subscript based keypaths, as
```